crates/runtime/benches/bench_search/mod.rs

```diff
@@ -14,5 +14,145 @@ See the License for the specific language governing permissions and
+
+    started_at: i64,
+    finished_at: i64,
+
+    index_started_at: i64,
+    index_finished_at: i64,
+
+    search_started_at: i64,
+    search_finished_at: i64,
+    search_response_time: Vec<f64>,
+
+    status: String,
+}
+
+impl Display for SearchBenchmarkResultBuilder {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "configuration: {},\n\
+          status: {},\n\
+          run_id: {},\n\
+          commit_sha: {},\n\
+          branch_name: {},\n\
+          started_at: {},\n\
+          index_time: {:.2} ms,\n\
+          search_time: {:.2} ms,\n\
+          rps: {:.2},\n\
+          mean_response_time: {:.2} ms,\n\
+          p95_response_time: {:.2} ms\n",
+            self.config_name,
+            self.status,
+            self.run_id,
+            self.commit_sha,
+            self.branch_name,
+            self.started_at,
+            self.index_finished_at - self.index_started_at,
+            self.search_finished_at - self.search_started_at,
+            self.rps(),
+            self.mean(),
+            self.quantile(0.95)
+        )
+    }
+}
+
+impl SearchBenchmarkResultBuilder {
+    pub fn new(
+        commit_sha: impl Into<String>,
+        branch_name: impl Into<String>,
+        config_name: impl Into<String>,
+    ) -> Self {
+        Self {
+            run_id: uuid::Uuid::new_v4().to_string(),
+            commit_sha: commit_sha.into(),
+            branch_name: branch_name.into(),
+            config_name: config_name.into(),
+            started_at: get_current_unix_ms(),
+            ..Default::default()
+        }
+    }
+
+    pub fn configuration_name(&self) -> String {
+        self.config_name.clone()
+    }
+
+    pub fn start_index(&mut self) {
+        self.index_started_at = get_current_unix_ms();
+    }
+
+    pub fn finish_index(&mut self) {
+        self.index_finished_at = get_current_unix_ms();
+    }
+
+    pub fn start_search(&mut self) {
+        self.search_started_at = get_current_unix_ms();
+    }
+
+    pub fn finish_search(&mut self) {
+        self.search_finished_at = get_current_unix_ms();
+    }
+
+    pub fn record_response_time(&mut self, response_time: f64) {
+        self.search_response_time.push(response_time);
+    }
+
+    pub fn finish(&mut self, is_successful: bool) {
+        self.finished_at = get_current_unix_ms();
+        self.status = if is_successful { "completed" } else { "failed" }.to_string();
+    }
+
+    // Calculate Requests Per Second (RPS)
+    #[allow(clippy::cast_precision_loss)]
+    pub(crate) fn rps(&self) -> f64 {
+        let total_time_sec =
+            (self.search_finished_at as f64 - self.search_started_at as f64) / 1000.0;
+        self.search_response_time.len() as f64 / total_time_sec
+    }
+
+    // Calculate Mean Value
+    #[allow(clippy::cast_precision_loss)]
+    pub(crate) fn mean(&self) -> f64 {
+        if self.search_response_time.is_empty() {
```

Comment:
Maybe this should `mean(&self) -> Option<f64> {` to avoid accidental use of 0.0.

Reply:
Good point, thank you - confirm this will be improved/updated as par of next PR